### DISCUSSION TITLE
iLogtail RoadMap

### DISCUSSION THREAD

**Discussion by messixukejia** [alibaba/ilogtail#422] [Announcements]
iLogtail开源以来，建立了很广泛的社区合作计划，现将一些规划与社区分享，期待一起开发共建。
我们也会持续更新该列表，欢迎就您关注的功能进行留言交流，我们将根据社区热度进行优先级的调整；如果您关心的特性不在该列表也欢迎留言，经评估采纳后也将被列入规划项目。

## 样例及模版

您可以从这些案例中快速了解iLogtail的功能，并贡献您的使用案例
https://github.com/alibaba/ilogtail/tree/main/example_config/user_contrib
- #279

## 基础功能

- [x] 时间精度支持 #952
- [x] GC高级特性（Go版本升级到1.19） #936
- [ ] C++/Golang处理链路打通，C++加速插件支持第三方输出 #855 #995
- [x] 多行切分模式升级 #978
- [ ] C++支持插件扩展
- [ ] 第三方输出可靠性：流量控制、数据可靠性框架

## 框架增强

- [x] 日志协议转换框架支持 @henryzhx8
- [x] 数据模型升级  Logs、Traces、Metrics通用性扩展 @liuhaoyang
- [ ] iLogtail自身可观测 #716 #1015

## 全局管控

- ConfigServer管控服务邀您共建 #917
- [x] 管控协议 @messixukejia
- [x] 管控服务 @Takuka0311
- [x] 管控前端交互 @liangry #894
- [ ] K8s Operator

## 生态支持

- [ ] 存量 Processor 插件迁移到 ProcessorV2 接口 #784

### Processor

- [x] Log to SLS Metric #949
- [ ] Log to General Metric
- [ ] Sampler

### Input

- [x] 新版Kafka Input  @shalousun
- [x] gRPC Input @shunjiazhu
- [x] HTTP Input 支持InfluxDB 
- [x] OTLP Log HTTP Input
- [x] OTLP Metric HTTP Input
- [x] OTLP Trace HTTP Input
- [x] OTLP Log GRPC Input
- [x] OTLP Metric GRPC Input
- [x] OTLP Trace GRPC Input
- [x] Command Input @didachuxing #925

### Output

- [x] OTLP Log Flusher
- [x] Loki Flusher
- [x] Kafka Flusher完善 @shalousun 
- [x] ClickHouse Flusher  @kl7sn @askuy
- [x] Elasticsearch Flusher
- [x] HTTP Flusher @snakorse
- [x] InfluxDB Flusher @snakorse
- [ ] gRPC Flusher完善
- [x] Pulsar Flusher @shalousun 
- [ ] OSS Flusher
- [ ] RocketMQ Flusher
- [ ] Doris Flusher

### OpenTelemetry Collector Receiver集成

待补充

### eBPF协议

- [x] L4、L7采集能力。
- [x] 支持HTTP、Redis、MySQL、PgSQL、DNS 流量观测。
- [x] 数据结构规范化&低版本内核
- [ ] L4 网络状态观测
- [ ] 用户态协议补充，如Dubbo、gRPC等
- [ ] Profling 支持

### Continuous Profiling

- [x] Golang
- [x] Java
- [x] Python
- [ ] eBPF
- [ ] ...

## 系统/架构支持

- [x] Linux X86
- [x] [Linux ARM](https://github.com/alibaba/ilogtail/pull/357)
- [x] [Windows](https://github.com/alibaba/ilogtail/issues/327)

## 部署模式

- [x] [Daemonset模式](https://github.com/alibaba/ilogtail/tree/main/k8s_templates)
- [x] [SiderCar模式](https://github.com/alibaba/ilogtail/tree/main/k8s_templates)
- [ ] 级联模式

**Comment by simops-328** [comment]


**Reply by messixukejia** [reply]
具体的需求场景能否详细描述下？

**Reply by simops-328** [reply]
我们希望在k8s里边使用一个agent采集Metric、logs以及ebpf的数据，当前ilogtail已经实现了Metric、logs的采集，但是ebpf的采集支持的协议比较少，希望支持多的协议，以及 Auto tracing等功能

**Reply by messixukejia** [reply]
但是ebpf的采集支持的协议比较少，希望支持多的协议 -- 期望支持的协议能否具体列一下？可以一起参与共建。

**Reply by simops-328** [reply]
> ebpf的协议比较，希望支持的协议——希望支持的协议能多列一下？可以一起参与建。

大体列出来就这些：http https DNS MySQL Oracle ElasticSearch Clickhouse kafka redis zookeeper oceanbase gRPC dobbo 各类数据库

**Reply by EvanLjp** [reply]
目前ebpf 在规范化数据协议，是否有兴趣一起交流下，支持协议是一个长期过程，可能每个用户的需求不同，欢迎进行基于规范化数据协议进行交流。

**Reply by messixukejia** [reply]
我们先提供下开发指南吧，下次双周会或者meetup讲一下。 @EvanLjp

**Reply by EvanLjp** [reply]
> 我们先提供下开发指南吧，下次双周会或者meetup讲一下。 @EvanLjp

目前这个的规范整理完了，涉及一些代码的抽象，进行中，还得2周吧，规范参考https://github.com/alibaba/ilogtail/pull/447

**Comment by gohonsen** [comment]
想问一下全局管控管控协议会包含哪些功能 ？会有停止采集/恢复采集的API 吗？

**Reply by messixukejia** [reply]
可以看下https://github.com/alibaba/ilogtail/blob/main/docs/cn/config-server/communication-protocol.md

可以通过采集配置跟Agent组进行绑定/解绑达到近似对效果。

**Comment by liuhaoyang** [comment]
> iLogtail开源以来，建立了很广泛的社区合作计划，现将一些规划与社区分享，期待一起开发共建。 我们也会持续更新该列表，欢迎就您关注的功能进行留言交流，我们将根据社区热度进行优先级的调整；如果您关心的特性不在该列表也欢迎留言，经评估采纳后也将被列入规划项目。
> 
> ## 样例及模版
> 您可以从这些案例中快速了解iLogtail的功能，并贡献您的使用案例 https://github.com/alibaba/ilogtail/tree/main/example_config/user_contrib
> 
> * [How to contribute | 如何参与开源贡献 #279](https://github.com/alibaba/ilogtail/issues/279)
> 
> ## 基础功能
> * [ ]  时间精度支持 [feat: support nanosecond #952](https://github.com/alibaba/ilogtail/pull/952)
> * [x]  GC高级特性（Go版本升级到1.19） [upgrade go.mod version to 1.19 #936](https://github.com/alibaba/ilogtail/pull/936)
> * [ ]  C++/Golang处理链路打通，C++加速插件支持第三方输出 [C++ Pipeline需求和演进 #855](https://github.com/alibaba/ilogtail/discussions/855) [C++ pipeline stage1: processors #995](https://github.com/alibaba/ilogtail/pull/995)
> * [ ]  多行切分模式升级 [feat: support continue/end multiple regex patterns #978](https://github.com/alibaba/ilogtail/pull/978)
> * [ ]  C++支持插件扩展
> * [ ]  第三方输出可靠性：流量控制、数据可靠性框架
> 
> ## 框架增强
> * [x]  日志协议转换框架支持 @henryzhx8
> * [x]  数据模型升级  Logs、Traces、Metrics通用性扩展 @liuhaoyang
> * [ ]  iLogtail自身可观测 [iLogtail observability enhancement #716](https://github.com/alibaba/ilogtail/discussions/716) [add metric new #1015](https://github.com/alibaba/ilogtail/pull/1015)
> 
> ## 全局管控
> * ConfigServer管控服务邀您共建 [ConfigServer管控服务邀您共建 #917](https://github.com/alibaba/ilogtail/discussions/917)
> * [x]  管控协议 @messixukejia
> * [x]  管控服务 @Takuka0311
> * [x]  管控前端交互 @liangry [doc: add config server ui quick start #894](https://github.com/alibaba/ilogtail/pull/894)
> * [ ]  K8s Operator
> 
> ## 生态支持
> * [ ]  存量 Processor 插件迁移到 ProcessorV2 接口 [存量 Processor 插件迁移到 ProcessorV2 接口 #784](https://github.com/alibaba/ilogtail/discussions/784)
> 
> ### Processor
> * [x]  Log to SLS Metric [Log to metric processor #949](https://github.com/alibaba/ilogtail/discussions/949)
> * [ ]  Log to General Metric
> * [ ]  Sampler
> 
> ### Input
> * [x]  新版Kafka Input  @shalousun
> * [x]  gRPC Input @shunjiazhu
> * [x]  HTTP Input 支持InfluxDB
> * [x]  OTLP Log HTTP Input
> * [x]  OTLP Metric HTTP Input
> * [x]  OTLP Trace HTTP Input
> * [x]  OTLP Log GRPC Input
> * [x]  OTLP Metric GRPC Input
> * [x]  OTLP Trace GRPC Input
> * [x]  Command Input @didachuxing [Support to collect cmd output： input_command(exec shell script from scriptcontent) #925](https://github.com/alibaba/ilogtail/pull/925)
> 
> ### Output
> * [x]  OTLP Log Flusher
> * [x]  Loki Flusher
> * [x]  Kafka Flusher完善 @shalousun
> * [x]  ClickHouse Flusher  @kl7sn @askuy
> * [x]  Elasticsearch Flusher
> * [x]  HTTP Flusher @snakorse
> * [x]  InfluxDB Flusher @snakorse
> * [ ]  gRPC Flusher完善
> * [x]  Pulsar Flusher @shalousun
> * [ ]  OSS Flusher
> * [ ]  RocketMQ Flusher
> * [ ]  Doris Flusher
> 
> ### OpenTelemetry Collector Receiver集成
> 待补充
> 
> ### eBPF协议
> * [x]  L4、L7采集能力。
> * [x]  支持HTTP、Redis、MySQL、PgSQL、DNS 流量观测。
> * [x]  数据结构规范化&低版本内核
> * [ ]  L4 网络状态观测
> * [ ]  用户态协议补充，如Dubbo、gRPC等
> * [ ]  Profling 支持
> 
> ### Continuous Profiling
> * [x]  Golang
> * [x]  Java
> * [x]  Python
> * [ ]  eBPF
> * [ ]  ...
> 
> ## 系统/架构支持
> * [x]  Linux X86
> * [x]  [Linux ARM](https://github.com/alibaba/ilogtail/pull/357)
> * [x]  [Windows](https://github.com/alibaba/ilogtail/issues/327)
> 
> ## 部署模式
> * [x]  [Daemonset模式](https://github.com/alibaba/ilogtail/tree/main/k8s_templates)
> * [x]  [SiderCar模式](https://github.com/alibaba/ilogtail/tree/main/k8s_templates)
> * [ ]  级联模式

感觉可以刷新一下Y24的RoadMap 呀？

**Reply by Takuka0311** [reply]
是的，我们近期刷新一下

**Comment by oldthreefeng** [comment]
why change name ilogtail to loongconllector ?
